json/src/main/java/io/airlift/json/JsonCodec.java

```
@@ -158,6 +161,31 @@ public String toJson(T instance)
     }
 
     /**
+     * Converts the specified instance to json string with a length limit.
+     *
+     * @param instance the instance to convert to json
+     * @param lengthLimit the maximum length of the serialized string
+     * @return json string
+     * @throws IllegalArgumentException if the specified instance can not be converted to json
+     * @throws JsonLengthLimitExceededException if the specified instance can not be converted to json because it exceeds the length limit
+     */
+    public String toJsonWithLengthLimit(T instance, int lengthLimit)
+            throws JsonLengthLimitExceededException
+    {
+        try (StringWriter stringWriter = new StringWriter();
+                LengthLimitedWriter lengthLimitedWriter = new LengthLimitedWriter(stringWriter, lengthLimit)) {
+            mapper.writeValue(lengthLimitedWriter, instance);
+            return stringWriter.getBuffer().toString();
+        }
+        catch (LengthLimitedWriter.LengthLimitExceededException e) {
+            throw new JsonLengthLimitExceededException(String.format("%s could not be converted to json string with length limit %d", instance.getClass().getName(), lengthLimit), e);
```

I don't think we need to propagate an internal exception as the cause